code/http/url/url.php

```diff
@@ -612,18 +624,18 @@ public function toString($parts = self::FULL, $escape = null)
             //Add the username and password
             if (($parts & self::USER) && !empty($this->user))
             {
-                $url .= urlencode($this->user);
+                $url .= rawurlencode($this->user);
```

Comment:
This could be double encoding the user.
Is there any way to set `$this->user` outside of the `setUser()` method? If not, then the `rawurlencode()` isn't needed.